### PR TITLE
[CARBONDATA-1212] Memory leak in case of compaction when unsafe is true

### PR DESCRIPTION
**Pull request by manishgupta88**
Problem: Memory leak in case of compaction when unsafe is true

Analysis: In case of compaction, queryExecutor object is formed for multiple blocks but the objects are not retained and finish method is called only on the last instance created for query executor. Due to this the memory allocated to precious objects is not released which can lead to out of memory issue.

Fix: Add all the object to a list and call finish method on all the objects so that memory gets released
Proof: Files attached for memory allocation and release count before and after code changes during compaction.
[Compaction_result_after_code_change.txt](https://github.com/apache/carbondata/files/1091871/Compaction_result_after_code_change.txt)
[Compaction_result_before_code_change.txt](https://github.com/apache/carbondata/files/1091872/Compaction_result_before_code_change.txt)

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2637/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/546/

File: processing/src/main/java/org/apache/carbondata/processing/merger/CarbonCompactionExecutor.java [hunk 1]
```diff
@@ -58,7 +58,7 @@
   private final Map<String, List<DataFileFooter>> dataFileMetadataSegMapping;
   private final SegmentProperties destinationSegProperties;
   private final Map<String, TaskBlockInfo> segmentMapping;
-  private QueryExecutor queryExecutor;
+  List<QueryExecutor> queryExecutorList;
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/merger/CarbonCompactionExecutor.java]: it should be private

**Comment by manishgupta88** [comment]
@jackylk ...handled review comment...please check

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2640/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/549/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**